### PR TITLE
Basic support for 12hours mode added

### PR DESCRIPTION
**Pull request by shaleynikov**

### CODE REVIEW

**Comment by afc163** [comment]
ref

https://github.com/ant-design/ant-design/issues/4063

#27

File: README.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ API
 | disabledHours           | Function                          | disabled hour options                         |                                                                                            |
 | disabledMinutes         | Function                          | disabled minute options                       |                                                                                            |
 | disabledSeconds         | Function                          | disabled second options                       |                                                                                            |
+| show12Hours             | Boolean                           | 12 hours display mode                         |                                                                                            |
```
Comment (afc163) [inline README.md]: show12Hours => use12Hours

**Comment by afc163** [comment]
image

I think it is better to show `04-31 PM` at top input.

File: src/Select.jsx [hunk 2]
```diff
@@ -58,7 +58,7 @@ const Select = React.createClass({
       });
       let onclick = null;
       if (!item.disabled) {
-        onclick = this.onSelect.bind(this, +item.value);
+        onclick = this.onSelect.bind(this, item.value);
```
Comment (benjycui) [inline src/Select.jsx]: Why?

I think it maybe a string.

Comment (shaleynikov) [inline src/Select.jsx]: It was explicitly converted by number, but that's not what I need handling 'AM/PM' selector

**Comment by shaleynikov** [comment]
@afc163 @benjycui shall we merge?

**Comment by afc163** [comment]
Could I set `format` to `HH:mm a` then input show like `04:30 am`, `HH:mm A` for `04:30 AM`?

**Comment by shaleynikov** [comment]
@afc163 sure, format is depends on moment, which supports 12hours quite well. You may check it on examples/12hours.js or take a look at new tests.

If you see something is wrong, just let me know

**Comment by afc163** [comment]
Now the input text is always `04:30 am` with lowercase `am`. What can we do if we want to display as uppercase `AM`?

**Comment by shaleynikov** [comment]
@afc163 
For instance we can use 'format' prop like that `format="h:mm A"` which will give us `4:30 AM`

**Comment by afc163** [comment]
@shaleynikov Yep, that is what most people would expect, which is not supported yet.

image

**Comment by afc163** [comment]
image

Also it makes input value and selected panel value looks different.

**Comment by shaleynikov** [comment]
Now AM/PM options lowercased or uppercased depending on format.
Added format demo displaying support of AM/PM in exact way you've mentioned.

Screenshot


@afc163

File: examples/12hours.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import React from 'react';
+import ReactDom from 'react-dom';
+
+import moment from 'moment';
+
+import TimePicker from 'rc-time-picker';
+
+const showSecond = false;
+const str = showSecond ? 'h:mm a' : 'h:mm a';
```
Comment (pifantastic) [inline examples/12hours.js]: These format strings are the same, no?

Comment (shaleynikov) [inline examples/12hours.js]: Yeah, updated

**Comment by shaleynikov** [comment]
Fixed everything and ready to merge

**Comment by afc163** [comment]
2.3.0

Thx @shaleynikov !

**Pull request merged**